### ISSUE TITLE
Restructuring strings is supported by TS but not by TSTL/Lua

### ISSUE THREAD

**Issue by lolleko** [TypeScriptToLua/TypeScriptToLua#1429]
Because strings are iterable, TS supports destructuring them into individual characters:

`const [a, b] = "test"; // a = "t" ; b = "e"`;

We currently do not support that in TSTL.
We should think about adding support for handling strings in restructuring assignments. At the very least, we should throw a diagnostic that this currently does not work.

Current behavior:

https://typescripttolua.github.io/play/#code/GYVwdgxgLglg9mABMOcAUUAWMDOAuRANzhgBMBKAnKAJxjAHNEBvAKEQ8RoFMoQakAIijdqggNysAvq1YQE1RAG0AhgBpEAIwC6iALzJUacpNYAHOmChp1Wk0A

Epected behaviour a = "t", b = "e";

https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABMOcAUUAWMDOAuRANzhgBMBKAnKAJxjAHNEBvAKEQ8RoFMoQakAIijdqggNysAvq1YQE1RAG0AhgBpEAIwC6iALzJUacpLkK4AG24A6C3AZp1Wk0A

Discovered while fixing #1411

**Comment by lolleko** [comment]
Related to #590

**Comment by Perryvw** [comment]
What is the proposed output for your example:
```lua
local __temp = foo()
local a = temp[0]
local b = temp[1]
```
Something like this?

**Comment by lolleko** [comment]
Modern JS (done know which standard maybe es5?) actually has built in support for string/iterator destructuring.

For older targets TS transpiles to what you recommended. So we can probably get away with treating strings the same as arrays/tuples.

**Comment by lolleko** [comment]
Yes for literal variable declarations, we can just treat them the same as array literals. However, for destructured assignment, we use `table.unpack` which wont work on string types.
